components-core/src/main/java/com/adyen/checkout/components/status/StatusRepository.java

```diff
@@ -134,6 +141,11 @@ public void updateStatus() {
      */
     public void stopPolling() {
         Logger.d(TAG, "stopPolling");
+        if (!mIsPolling) {
```

Comment:
I'd rather `stopPolling` happen regardless of the flag value. If the flag goes to an inconsistent state for some reason, at least stop would force a reset.